### DISCUSSION TITLE
Wifi connection not enabled by watchdog - PRO CPU has been reset by WDT

### DISCUSSION THREAD

**Discussion by gkzsolt** [micropython/micropython#10377] [ESP32]
I've been using Micropython succesfully with a few ESP8266 and ESP32 boards for a couple of years. Now I cannot connect to Wifi from my newly acquisitioned NodeMCU-ESP-32S boards. The following code:

```
import network
wlan = network.WLAN(network.STA_IF)
wlan.active(True)
wlan.scan()   # or wlan.connect('your-ssid', 'your-key')
```
resets the board almost instantly:
```
ets Jul 29 2019 12:21:46

rst:0x8 (TG1WDT_SYS_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:4540
ho 0 tail 12 room 4
load:0x40078000,len:12344
ho 0 tail 12 room 4
load:0x40080400,len:4124
entry 0x40080680
W (61) boot.esp32: PRO CPU has been reset by WDT.
W (61) boot.esp32: WDT reset info: PRO CPU PC=0x40093e52
W (63) boot.esp32: WDT reset info: APP CPU PC=0x4008d3a2
```

Important: this happens with the following chip: `ESP32D0WDQ5 (revision 3)`. But it DOES WORK with `ESP32D0WDQ6 (revision 1)` (on the same NodeMCU-ESP-32S board type).

The MicroPython was v1.19.1 - but I tried all versions available at the [download page](https://micropython.org/download/esp32/), with firmware compiled with ESP-IDF v3.x or v4.x. Up to [v1.11](https://micropython.org/resources/firmware/esp32-idf3-20190529-v1.11.bin) the code just block without returning; higher versions all were resetting the board.

One more info: I flushed the boards also with C/C++ compiled code from Arduino IDE 1.8, and Wifi worked like a charm. So it seems to be not a hardware issue.

What can I do? Can you point me to some direction? I like Micropython and using it in quite large projects.

Thanks for your help!

Zsolt

**Comment by davefes** [comment]
Did you also try with the SPIRAM firmware variant? 

I'd try removing the scan() and just try blinking an LED.  Then try the do_connect() example in the docs.

**Reply by gkzsolt** [reply]
Hi Dave,

Thanks. Why should it work with the SPIRAM firmware? The board does not have SPIRAM, and we should upload the firmware most compatible with the device. I uploaded [this](https://micropython.org/download/esp32spiram/), but when trying to connect the REPL through USB I got an infinite restarting loop. Am I wrong supposing that the [generic firmware](https://micropython.org/download/esp32/), which worked with the version 1 board should work with the version 3 as well?

> I'd try removing the scan() and just try blinking an LED. 

I still don't get the connection with my issue, but I tried it and it works (with the generic ESP32 variant).

> Then try the do_connect() example in the docs.

If you take a peek at the do_connect() example, it is exactly the same what I tried, just in a loop. 

Regards,
  Zsolt

**Reply by davefes** [reply]
I was just suggesting that maybe it might be a spiram variant. I have rev1 and rev3 chips in esp32 wroom  dev boards with the same firmware.

**Reply by gkzsolt** [reply]
> I was just suggesting that maybe it might be a spiram variant. I have rev1 and rev3 chips in esp32 wroom dev boards with the same firmware.

Than why have you suggested to use an spiram variant? You have the same dev boards working with v1.19.1 (2022-06-18) .bin, right?

**Reply by davefes** [reply]
I did not know for certain that DOWDQ5 was the same variant as the DOWDQ6.  In order to verify what variant (DOWDQ?) the rev3 chips I have are I would need to erase them so that esptools would tell me what variant and revision they are.

 There also was some discussion where I believe someone said that they could run generic on the SPIRAM variant.

I am sorry if I confused things by suggesting something that may not have been helpful.

**Comment by gkzsolt** [comment]
Which firmware are you using?

**Reply by davefes** [reply]
v1.19.1 (2022-06-18) .bin

**Comment by TKrom** [comment]
I have the exact same problem.

Did you manage to find a solution so far by any chance @davefes ?

Best,
Teun

**Comment by benlye** [comment]
Same problem here with WDT reset when trying to scan or connect in station mode with an ESP-WROOM-32.  esptool reports `Chip is ESP32-D0WD-V3 (revision v3.0)`.

I'm erasing before flashing the latest (1.20.0) bin
```
C:\Temp> esptool.py --port COM4 erase_flash
esptool.py v4.6.2
Serial port COM4
Connecting...........
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting..........
Detecting chip type... ESP32
Chip is ESP32-D0WD-V3 (revision v3.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: b8:d6:1a:b2:71:20
Uploading stub...
Running stub...
Stub running...
Erasing flash (this may take a while)...
Chip erase completed successfully in 0.4s
Hard resetting via RTS pin...

C:\Temp> esptool.py --chip esp32 --port COM4 --baud 460800 write_flash -z 0x1000 esp32-20230426-v1.20.0.bin
esptool.py v4.6.2
Serial port COM4
Connecting.........
Chip is ESP32-D0WD-V3 (revision v3.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: b8:d6:1a:b2:71:20
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size...
Flash will be erased from 0x00001000 to 0x0017ffff...
Compressed 1566528 bytes to 1034676...
Wrote 1566528 bytes (1034676 compressed) at 0x00001000 in 23.9 seconds (effective 524.0 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
C:\Temp>
```

The when I use REPL to try to start Wi-Fi:
```
ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:4656
load:0x40078000,len:13284
ho 0 tail 12 room 4
load:0x40080400,len:3712
entry 0x4008064c
MicroPython v1.20.0 on 2023-04-26; ESP32 module with ESP32
Type "help()" for more information.
>>> import network
>>> sta_if = network.WLAN(network.STA_IF); sta_if.active(True)
True
>>> sta_if.scan()
ets Jul 29 2019 12:21:46

rst:0x7 (TG0WDT_SYS_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:4656
load:0x40078000,len:13284
ho 0 tail 12 room 4
load:0x40080400,len:3712
entry 0x4008064c
W (58) boot.esp32: PRO CPU has been reset by WDT.
W (58) boot.esp32: WDT reset info: PRO CPU PC=0x40093e2b
W (59) boot.esp32: WDT reset info: APP CPU PC=0x40094216
```

Like others, Arduino code can start Wi-Fi and scan/connect without issues, which seems to rule out a hardware issue.  Even so, I've tried multiple USB cables, ports, hubs, and computers. I've tried using the USB port and using an external FTDI adapter.

Is there any solution for this?

**Comment by benlye** [comment]
For the benefit of others who end up here - I fixed the MicroPython Wi-FI WDT reset problem on my ESP-WROOM-32 by adding a 10uF capacitor between `EN` and `GND`.  I came across this solution in https://github.com/orgs/micropython/discussions/10521.

The capacitor has the added benefit of making uploads using esptool.py no longer need me to press the buttons on the board.

![IMG20230713145057](https://github.com/micropython/micropython/assets/23322579/d342c127-4b94-42cb-a1ff-5fa67b378eeb)
